src/Native/System.Native/pal_utilities.cpp

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "../config.h"
+#include "pal_utilities.h"
+#include <string.h>
+#include <stdio.h>
+
+int32_t SafeStringCopy(char* destination, int32_t destinationSize, char* source)
+{
+#if HAVE_SAFE_STRCPY
+    return strcpy_s(destination, destinationSize, source);
+#elif HAVE_STRCPY_WITH_LENGTH
+    return strlcpy(destination, source, destinationSize);
+#else
+    return snprintf(destination, destinationSize, "%s", source);
+#endif
+}
```

Comment:
I don't think those return values all have the same semantics: strcpy_s returns errno_t while strlcpy and snprintf return the size they needed IIRC. I would just make the wrapper return void so that we don't have to deal with those differences.